### DISCUSSION TITLE
Prisma 5.9.0 wrong types with moduleResolution: bundler

### DISCUSSION THREAD

**Discussion by JohanHjelsethStorstad** [prisma/prisma#22901] [Q&A] [answered]
Hi.
I am having a problem with prisma 5.9.0. We have a next 14 app using prisma. It seams to get all the types wrong. For example when we have:  
```javascript 
const images = await prisma.image.findMany({
            where: {
                collectionId,
            },
            skip: pageNumber * pageSize,
            take: pageSize,
 })
```
The return type used to be Image on 5.8.1, now on 5.9.0 it's a generic $Public.PrismaPromise<T>, in other calls to prisma the return type can even be any.

The issue is completely resolved when I switch to moduleResolution: node. Does anyone know what is going on?

Thanks

**Comment by millsp** [comment, marked as answer]
Hey @JohanHjelsethStorstad, thanks for the report. This looks like [this issue](https://github.com/prisma/prisma/issues/22870), which we have a fix prepared for but not yet released.

Could you please try `5.10.0-integration-fix-client-exports-map.11` (change both your `prisma` and `@prisma/client`) version (don't forget to rerun `prisma generate` after switching version), and let me know if that fixes your issue?

We will do a patch release soon with the fixes.

**Reply by JohanHjelsethStorstad** [reply]
Thanks for the quick reply. This seams to solve it.
I will just wait for a patch

**Reply by vignesh-gupta** [reply]
@millsp  seems like the same error is happening in 6.1.0 again. 
![image](https://github.com/user-attachments/assets/86b691d7-5ece-4b28-9c28-342892d5d803)

**Reply by andreibsk** [reply]
I started happening for me after upgrading from `6.2.1` to `6.3.0`.
